src/init/power.rs

```diff
@@ -0,0 +1,122 @@
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use log::{error, info, trace};
+use std::{
+    fs::{OpenOptions, File},
+    io::Read,
+    mem,
+    path::Path,
+    slice,
+};
+
+extern crate libc;
+
+#[allow(dead_code)]
+fn syscall_reboot(action: i32) {
+    unsafe {
+        libc::reboot(action);
+    }
+}
+
+pub fn power_off() {
+    syscall_reboot(libc::LINUX_REBOOT_CMD_POWER_OFF);
+}
+
+#[allow(dead_code)]
+pub fn reboot() {
+    syscall_reboot(libc::LINUX_REBOOT_CMD_RESTART);
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+#[repr(C, packed)]
+pub struct InputEvent {
+    tv_sec: u64,
+    tv_usec: u64,
+    evtype: u16,
+    code: u16,
+    value: u32,
+}
+
+// see  https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/input-event-codes.h#L191
+const KEY_POWER: u16 = 116;
+
+#[allow(dead_code)]
+pub fn spawn_power_button_listener() {
```

Comment:
Can we be more specific about the term "spawn" please? 

```rust
pub fn spawn_thread_power_button_listener();
```

Maybe something like this? I just want to make sure we have a good convention to understand the different between `std::process:Command:Spawn` and `std::thread::Spawn` or even the `tokio::runtime::Spawn`.